url29d9f49ba9_frankrefischer_STTRJ_tgz-pJ8-startrekj_STTR1J8/src/startrekj/STTR1.java

```diff
@@ -0,0 +1,865 @@
+package startrekj;
+
+import static startrekj.hpbasic.Line.*;
+import static startrekj.hpbasic.functions.ABS.*;
+import static startrekj.hpbasic.functions.ADD.*;
+import static startrekj.hpbasic.functions.DIV.*;
+import static startrekj.hpbasic.functions.INT.*;
+import static startrekj.hpbasic.functions.MULT.*;
+import static startrekj.hpbasic.functions.RND.*;
+import static startrekj.hpbasic.functions.SQR.*;
+import static startrekj.hpbasic.functions.SUBTRACT.*;
+import static startrekj.hpbasic.functions.TIM.*;
+import static startrekj.hpbasic.Arithmetic.*;
+import static startrekj.hpbasic.operators.EQ.*;
+import static startrekj.hpbasic.operators.EQ_String.*;
+import static startrekj.hpbasic.operators.GT.*;
+import static startrekj.hpbasic.operators.GTE.*;
+import static startrekj.hpbasic.operators.LT.*;
+import static startrekj.hpbasic.operators.LTE.*;
+import static startrekj.hpbasic.operators.NEQ.*;
+import static startrekj.hpbasic.operators.NEQ_String.*;
+import static startrekj.hpbasic.operators.OR.*;
+import static startrekj.hpbasic.operators.SUBSTRING.*;
+import static startrekj.hpbasic.statements.SET_ArrayPlace.*;
+import static startrekj.hpbasic.statements.SET_StringVariable.*;
+import static startrekj.hpbasic.statements.SET_NumericVariable.*;
+import static startrekj.hpbasic.statements.DEF.*;
+import static startrekj.hpbasic.statements.DIM.*;
+import static startrekj.hpbasic.statements.DIM$.*;
+import static startrekj.hpbasic.statements.EXIT.*;
+import static startrekj.hpbasic.statements.FOR.*;
+import static startrekj.hpbasic.statements.GOSUB.*;
+import static startrekj.hpbasic.statements.GOTO.*;
+import static startrekj.hpbasic.statements.IF.*;
+import static startrekj.hpbasic.statements.IMAGE.*;
+import static startrekj.hpbasic.statements.INPUT.*;
+import static startrekj.hpbasic.statements.INPUT_String.*;
+import static startrekj.hpbasic.statements.MAT.*;
+import static startrekj.hpbasic.statements.NEXT.*;
+import static startrekj.hpbasic.statements.PRINT.*;
+import static startrekj.hpbasic.statements.PRINT_USING.*;
+import static startrekj.hpbasic.statements.REM.*;
+import static startrekj.hpbasic.statements.REPLACE.*;
+import static startrekj.hpbasic.statements.RETURN.*;
+import static startrekj.hpbasic.statements.TRACE.*;
+import startrekj.hpbasic.ArrayVariable;
+import startrekj.hpbasic.FunctionVariable;
+import startrekj.hpbasic.HPBasicProgram;
+import startrekj.hpbasic.NumericExpression;
+import startrekj.hpbasic.NumericVariable;
+import startrekj.hpbasic.StringVariable;
+import startrekj.hpbasic.statements.TRACE;
+
+public class STTR1 extends HPBasicProgram {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This class references 50 other classes. By comparison, 99% of the classes in the CodeGuru reference dataset reference fewer. This indicates that this class is highly coupled with other classes. A class that is highly coupled with other classes is difficult to understand and its behavior might change unexpectedly when one of its referenced classes is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt. We recommend that you simplify this class or break it into multiple classes.